src/server/search/doc_accessors.cc

```diff
@@ -106,13 +109,35 @@ SearchDocData StringMapAccessor::Serialize(const search::Schema& schema) const {
 struct JsonAccessor::JsonPathContainer : public jsoncons::jsonpath::jsonpath_expression<JsonType> {
 };
 
-string_view JsonAccessor::GetString(string_view active_field) const {
-  auto res = GetPath(active_field)->evaluate(json_);
-  DCHECK(res.is_array());
-  if (res.empty())
-    return "";
-  buf_ = res[0].as_string();
-  return buf_;
+BaseAccessor::StringList JsonAccessor::GetStrings(string_view active_field) const {
+  auto path_res = GetPath(active_field)->evaluate(json_);
+  DCHECK(path_res.is_array());  // json path always returns arrays
+
+  if (path_res.empty())
+    return {};
+
+  if (path_res.size() == 1) {
+    buf_ = path_res[0].as_string();
+    return {buf_};
+  }
+
+  StringList out;
+
+  // First, grow buffer and compute string sizes
+  for (auto element : path_res.array_range()) {
+    size_t start = buf_.size();
+    buf_ += element.as_string();
+    out.push_back(string_view{nullptr, buf_.size() - start});
```

Comment:
This is an UB, I'm afraid. `string_view` must point at a valid range: https://timsong-cpp.github.io/cppwp/string.view.cons
Maybe start with a `vector<pair<size_t, size_t>>` and then convert to `StringList`? I realize it would require another allocation, but it's probably better to not have UB

Reply:
True, strange requirement for such a dead simple type... Fixed